language-support/java/bindings-rxjava/src/main/java/com/daml/ledger/rxjava/components/helpers/TemplateUtils.java

```diff
@@ -0,0 +1,49 @@
+// Copyright (c) 2019 The DAML Authors. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package com.daml.ledger.rxjava.components.helpers;
+
+import com.daml.ledger.javaapi.data.Identifier;
+import com.daml.ledger.javaapi.data.Record;
+import com.daml.ledger.javaapi.data.Template;
+import com.daml.ledger.javaapi.data.Value;
+import com.daml.ledger.rxjava.components.helpers.CreatedContract;
+import java.lang.reflect.InvocationTargetException;
+import java.util.function.Function;
+
+public class TemplateUtils {
+
+    /**
+     * Creates a transform function which is aware of the template types. Useful to be passed to
+     * Bot.wire.
+     *
+     * @param allowedTemplates the list of allowed template types
+     * @return a transformation function returning a new contract typed as one of the allowed
+     *     templates
+     */
+
+    public static Function<CreatedContract, Template> contractTransformator(
+            Class<? extends Template>... allowedTemplates) {
+        return createdContract -> {
```

Comment:
@gerolf-da how hard would it be to generate this code, and instead of using reflection, generate a big `if` statement?

Reply:
In fact, the java codegen can generate a "Decoder" class that does the matching via the template ID and returns a statically known function, so no reflection is needed. See [decoder docs](https://docs.daml.com/app-dev/bindings-java/codegen.html#generate-the-decoder-utility-class) for a bit more information.

Reply:
But that would have to happen in the user code, as this generic utility does not have access to the statically generated classes.

Reply:
Mmmmh, so we had this already? /cc @richardkapolnai-da

Reply:
Ok, totally my fault for not recognizing this was the decoder. I'll improve the documentation for it.

Reply:
https://github.com/digital-asset/daml/pull/3124

Reply:
Funny. I also could have read the docs... :)
However, I made a comment on the other PR, the equality is not trivial to me.